open_ce/build_image.py

```diff
@@ -49,32 +51,48 @@ def build_image(local_conda_channel, conda_env_file, container_tool, container_b
     """
     variant = os.path.splitext(conda_env_file)[0].replace(utils.CONDA_ENV_FILENAME_PREFIX, "", 1)
     variant = variant.replace("-runtime", "")
+
+    # Docker version on ppc64le rhel7 doesn't allow Dockerfiles to be out of build context.
+    # Hence, copying it in temp_dir inside the build context. This isn't needed with newer 
+    # docker versions or podman needed but to be consistent, doing this in all cases.
+    dockerfile_path = os.path.join(local_conda_channel, TEMP_FILES, "Dockerfile")
+    runtime_img_file = _get_runtime_image_file(container_tool)
+    create_copy(runtime_img_file, dockerfile_path)
+ 
     image_name = REPO_NAME + ":" + IMAGE_NAME + "-" + variant
     build_cmd = container_tool + " build "
-    build_cmd += "-f " + os.path.join(RUNTIME_IMAGE_PATH, "Dockerfile") + " "
+    build_cmd += "-f " + dockerfile_path + " "
     build_cmd += "-t " + image_name + " "
     build_cmd += "--build-arg OPENCE_USER=" + OPENCE_USER + " "
-    build_cmd += "--build-arg LOCAL_CONDA_CHANNEL=" + local_conda_channel + " "
-    build_cmd += "--build-arg CONDA_ENV_FILE=" + conda_env_file + " "
+    build_cmd += "--build-arg LOCAL_CONDA_CHANNEL=" + "./ "
+    build_cmd += "--build-arg CONDA_ENV_FILE=" + os.path.join(TEMP_FILES, conda_env_file) + " "
     build_cmd += "--build-arg TARGET_DIR=" + TARGET_DIR + " "
     build_cmd += container_build_args + " "
-    build_cmd += BUILD_CONTEXT
+    build_cmd += local_conda_channel
 
     print("Container build command: ", build_cmd)
     if os.system(build_cmd):
         raise OpenCEError(Error.BUILD_IMAGE, image_name)
 
     return image_name
 
-def _validate_input_paths(local_conda_channel, conda_env_file):
+def _get_runtime_image_file(container_tool):
+    tool_ver = utils.get_container_tool_ver(container_tool)
+    image_file = os.path.join(RUNTIME_IMAGE_PATH, "docker/Dockerfile")
+    if not tool_ver:
+        print("WARNING: Could not retrieve version of {} container tool".format(container_tool))
+        # If we couldn't get the version of the tool, should use the older docker
+        # supported Dockerfile
+        return image_file
 
-    # Check if path exists
-    if not os.path.exists(local_conda_channel) or not os.path.exists(conda_env_file):
-        raise OpenCEError(Error.INCORRECT_INPUT_PATHS)
+    tool_ver = tool_ver.replace(".", "")
+    if (container_tool == "docker" and int(tool_ver) >= 1709) or \
```

Comment:
For docker I read that `COPY --chown` was introduced in 17.09 version. With podman I could not check if `COPY --chown` is supported since beginning or with versions >= 2.0. We have podman 2.0.5 on rhel8(ppc and x86) where this cmd works. On dlw07, we have podman 1.6.4 but I'm unable to test on it due to some error. And with root user, conda setup isn't working.
@dllehr81 - Any idea if this feature has been there in podman since beginning? If yes, I'll just remove the version check from this condition.